spec/classes/issue_spec.rb

```diff
@@ -1,90 +1,83 @@
 require 'spec_helper'
 
 describe 'issue', type: :class do
-  describe 'On Linux' do
-    let(:facts) do
-      {
-        kernel: 'Linux',
-        operatingsystem: 'TestOS',
-        osfamily: 'Debian',
-        memoryfree: '1 KB',
-        domain: 'testdomain'
-      }
-    end
-    context 'When nothing is specified' do
-      it do
-        should contain_File('/etc/issue').with(
-          ensure: 'file',
-          content: /ATTENTION/
-        )
-        should contain_File('/etc/issue.net').with(
-          ensure: 'file',
-          source: 'file:///etc/issue'
-        )
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      context 'When nothing is specified' do
+        it do
+          should contain_File('/etc/issue').with(
+            ensure: 'file',
+            content: /ATTENTION/
+          )
+          should contain_File('/etc/issue.net').with(
+            ensure: 'file',
+            source: 'file:///etc/issue'
+          )
+        end
```

Comment:
Would like to see a test with one of the new banner names to ensure that logic path is tested.